### ISSUE TITLE
Regarding your Experiments in 5.3. FashionMNIST with corrupted labels

### ISSUE THREAD

**Issue by sdogsq** [uuujf/SGDNoise#1]
Hi,

Recently I am following your work The Anisotropic Noise in Stochastic Gradient Descent: Its Behavior of Escaping from Sharp Minima and Regularization Effects and have been doing an experiment similar to **5.3. FashionMNIST with corrupted labels.**

I am planning to reproduce this work. At beginning, I programmed a PyTorch version of this experiment following the training details in D.3. FashionMNIST with corrupted labels. But I cannot reproduce the result in Fig.1. The result given by my code is shown in Fig. 2. In this part, I have two questions:

1. The test accuracy keeps at around 0.70 (refer to line GD pre 0.1 in Fig.2) after pretraining by **GD, which is far higher than that in Fig.1 (0.6622).**

2. Test accuracy is much lower than GD after keeping training by SGD (refer to line MSGD 0.07 20 in Fig.2). **It does not grow up but drops a lot instead, which is against to Fig.1.**

I run my code several times but unfortunately above questions keep showing up.

![image](https://user-images.githubusercontent.com/4051226/94393563-eb9ec700-018d-11eb-8cce-914a238e2a6a.png)
(Fig. 1)

![image](https://user-images.githubusercontent.com/4051226/94393572-f22d3e80-018d-11eb-97ea-e31abb045193.png)
(Fig. 2)

Later on, I found this repo. But I still have a few questions.

1. In the paper, 1200 samples in total are used to train but in this code **it generates 6000 samples in training dataset(Fig.3).** Please refer to [line 80(click here to github website)](https://github.com/uuujf/SGDNoise/blob/b21ad2dcab1bb2833bb16991e81f702c322acbb3/FashionMNIST/mnist2np.py#L80). I think it is just a typo and change it to 1200.

2. I ran gd.py and sgd.py after I change it to 1200. I tried several times but **I still meet the problem above as my own code(Fig.4).**

![image](https://user-images.githubusercontent.com/4051226/94393574-f6f1f280-018d-11eb-8176-6e422f955d76.png)
(Fig.3)

![image](https://user-images.githubusercontent.com/4051226/94393584-f9ece300-018d-11eb-861c-4d7565d3c2de.png)
(Fig.4)

I really appreciate your insightful work and innovative idea. But now I really have no idea about this experiment. Would you please give me some advice? Thank you a lot for your help and I am looking forward to hearing from you soon!

Best wishes!

**Comment by XinyuanWang283** [comment]
Hello, Sir, have you solved this problem? Could I send you an email to talk about this problem?

**Comment by sdogsq** [comment]
> Hello, Sir, have you solved this problem? Could I send you an email to talk about this problem?

It has been quite a long time since I posted this issue. After this issue I sent an email to the authors and they offered me a pytorch implement. Perhaps you could also send them an email.  : )
